zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/clock/DbClockState.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.clock;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.engine.state.mutable.MutableClockState;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.stream.api.StreamClock.ControllableStreamClock.Modification;
+
+public class DbClockState implements MutableClockState {
+  private static final DbString KEY = new DbString("MODIFICATION");
```

Comment:
:wrench: while it's _probably_ fine for a `DbString`, I'd caution against static fields here. In the past we've had issues where writing such a static value would surprisingly mutate internal state, causing race conditions across partitions.

Maybe just do it how it's done in `DbCheckpointState`.

Reply:
Wait, if I _only_ use it in an upsert, it might be mutated? :exploding_head:

Reply:
It looks like `DbString` is safe but messagepack values aren't, their `writer` is mutating internal state while writing. Writing may occur on `upsert` because RocksDB needs the raw bytes so we write.

Reply:
I could make an immutable type specifically for this, but I think that's only useful if there were plans to use it in other places, and while I think it's probably useful, I can't tell for sure :sweat_smile: